syntactic-paths/src/main/java/com/palantir/util/syntacticpath/Path.java

```diff
@@ -131,7 +120,10 @@ private Path normalizeInternal() {
      * Backward navigation in empty paths is a no-op, e.g., {@code a/../..} normalizes to the empty path.
      */
     public Path normalize() {
-        return normalizedPath.get();
+        if (lazyInitializedNormalizedPath == null) {
```

Comment:
well, it might not need to be because the assignment is atomic and idempotent? I just want you to think about it and document accordingly :)